addons/website/views/website_templates.xml

```diff
@@ -155,16 +155,28 @@
     <!-- Page options -->
     <xpath expr="//div[@id='wrapwrap']" position="before">
         <t groups="website.group_website_publisher">
-            <t t-foreach="['header_overlay', 'header_color', 'header_visible', 'footer_visible']" t-as="optionName">
+            <t t-foreach="['header_overlay', 'header_color', 'header_text_color', 'header_visible', 'footer_visible']" t-as="optionName">
                 <input t-if="optionName in main_object" type="hidden" class="o_page_option_data" t-att-name="optionName" t-att-value="main_object[optionName]"/>
             </t>
         </t>
+        <!-- background and text color applied for only transparent header -->
+        <t t-set="header_bg_color" t-value="'header_color' in main_object and main_object.header_color"/>
+        <t t-set="header_text_color" t-value="'header_text_color' in main_object and main_object.header_text_color"/>
     </xpath>
     <xpath expr="//div[@id='wrapwrap']" position="attributes">
         <attribute name="t-attf-class" add="#{'o_header_overlay' if 'header_overlay' in main_object and main_object.header_overlay else ''}" separator=" "/>
     </xpath>
+    <!-- for transparent header background color -->
```

Comment:
`<!-- for overlay header background color -->`

Reply:
changes have done!